### PR TITLE
adding balloon transport interface

### PR DESCRIPTION
**Pull request by Reldnahc**
I'm not really sure if I did this right. Let me know and I can make changes accordingly.  Everything works in each client mode.

### CODE REVIEW

File: osr/interfaces/balloontransport.simba [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+  TRSBalloonTransport = type TRSInterface;
+
+procedure TRSBalloonTransport.Setup(); override;
+begin
+  inherited;
+  ButtonColors := [CTS1(8564948, 0)];
+end;
+
+procedure TRSBalloonTransport.SetupAlignment(mode: ERSClientMode); override;
```
Comment (Reldnahc) [inline osr/interfaces/balloontransport.simba:19]: I don't really understand this part but it works in all three modes.}

Resize classic
image

Resize modern
image

Fixed
image

**Comment by Torwent** [comment]
The alignment is like an anchor point. without it resizable wouldn't work properly because your can resize the window.
It basically makes coordinates dynamic in relation to the anchor points.

It goes look good. Good job.
quick question before taking a better look and merging it, are the click boxes right?

I ask because Grand tree and Castle wars seem to have a lot of empty space to the right. Is that empty space clickable?
It wouldn't be something new, rs has all sort of weird clickboxes but I figured I would ask to be sure!

**Comment by Reldnahc** [comment]
Thanks! They are correct, I tried to make them all just a little smaller than the clickable area on the in-game interface.

**Comment by Torwent** [comment]
Hey, see if you like my changes, if you are alright with them I'll merge this

**Comment by Torwent** [comment]
To put it simple, I changed so GetButtons use GetButton and get button is the one that has a manual setup for each button.

It didn't make sense imo to manually set each button on `.GetButtons()` and then having to loop each of them to get a single one with `.GetButton()`.
With the new way, as far as `.GetButtons()` is concerned it takes the same amount of computation, but `.GetButton()` is faster because it doesn't require looping.

**Comment by Reldnahc** [comment]
The changes you made are great. I agree it didn't make sense how I had it before. I was guilty of copy-pasting and not thinking about what it was actually doing.

**Pull request merged**